graph_explorer/src/graph_explorer/graph_visualizer/views.py

```diff
@@ -133,6 +133,78 @@ def delete_filter(request):
 
     return HttpResponse(200, content_type="application/json")
 
+
+def edit_workspace(request, id, datasource_name=None):
+    data_sources = get_datasource_names()
+    ws: Workspace = app_config.get_workspace(id)
+    if ws is None:
+        return HttpResponseNotFound("Workspace with given id not found.")
+    datasource_name = ws.selected_datasource if datasource_name is None else datasource_name
+
+    datasource_config_params = get_datasource_configuration(datasource_name)
+    ds_config_params = datasource_config_params.items()
+    ds_config_params_with_values = []
+    for key, key_type in ds_config_params:
+        value = ""
+        if ws is not None and key in ws.datasource_config:
+            value = ws.datasource_config[key]
+        ds_config_params_with_values.append((key, key_type, value))
```

Comment:
This for loop should only be executed when the datasource_name is the initial data source, in this case, two data source plugins may have the same parameter key name so it is persisted when changing the data source plugin. This might be a feature also, decide if you want to change it. Nothing urgent.